app/controllers/api/v1/recipes_controller.rb

```diff
@@ -1,36 +1,41 @@
 class Api::V1::RecipesController < ApplicationController
   before_action :authenticate_user!, :except => [:show, :index]
 
-  def show
-    @recipe = Recipe.find_by!(id: params[:id])
-    render json: @recipe, :include => [{comments: {include: :user }}, :tags]
-  end
+    def show
+      @recipe = Recipe.find_by!(id: params[:id])
+      render json: @recipe, :include => [{comments: {include: :user }}, :tags]
+    end
 
-  def index
-    if params[:search]
-      @recipes = Recipe.search(params[:search])
-    else
-      @recipes = Recipe.includes(:user).all
+    def index
+      if params[:search]
+        @recipes = Recipe.search(params[:search])
+      else
+        if params[:user_id]
```

Comment:
I would try to avoid nesting if statements unless there's a good reason. In this case, you could just use [`elsif`](https://www.tutorialspoint.com/ruby/ruby_if_else.htm).